internal/handlers/common/update.go

```diff
@@ -118,5 +122,77 @@ func UpdateDocument(doc, update *types.Document) (bool, error) {
 		}
 	}
 
-	return true, nil
+	return changed, nil
+}
+
+// ValidateUpdateQuery validates update statement.
+func ValidateUpdateQuery(update *types.Document) (err error) {
+	set, err := getDocument("$set", update)
+	if err != nil {
+		return err
+	}
+	_, err = getDocument("$setOnInsert", update)
+	if err != nil {
+		return err
+	}
+
+	inc, err := getDocument("$inc", update)
+	if err != nil {
+		return err
+	}
+
+	if set != nil && inc != nil {
+		for _, setKey := range set.Keys() {
+			if inc.Has(setKey) {
+				return NewWriteErrorMsg(
+					ErrConflictingUpdateOperators,
+					fmt.Sprintf("Updating the path '%[1]s' would create a conflict at '%[1]s'", setKey),
+				)
+			}
+		}
+	}
+
+	for _, updateOp := range update.Keys() {
+		switch updateOp {
+		case "$set",
+			"$setOnInsert",
+			"$inc":
+			// supported
+		default:
+			return NewWriteErrorMsg(
+				ErrFailedToParse,
+				fmt.Sprintf("Unknown modifier: %s. Expected a valid update modifier or pipeline-style "+
+					"update specified as an array", updateOp,
+				))
+		}
+	}
+	return nil
+}
+
+// getDocument gets document by key `op` and returns WriteError error if it is not a document.
+func getDocument(op string, update *types.Document) (*types.Document, error) {
```

Comment:
`getDocument` looks very common name for that function. Maybe something like `getUpdateOperatorDocument`?

Reply:
renamed